### PR TITLE
Service refactor

### PR DESCRIPTION
**Pull request by tesk9**
Creates bar_service to aid the lovely top bar nav directives.

Will need to test as a student and as an admin (must be user to be an admin).

### CODE REVIEW

**Comment by krish-andres** [comment]
You should remove the spaces in the conditionals and for loops (in your admin_top_nav_directive file).  Other than that looks great!

File: assets/js/directives/admin_top_nav_directive.js [hunk 1]
```diff
@@ -27,31 +25,11 @@ app.directive("adminTopNavDirective", ["$location", "$http", "$route", "AuthServ
             }
             //first item in list should be core or core should be displayed on nav bar
             if (scope.userArr[0].name !== "Core") {
-              scope.userArr.unshift({id:1,name:"Core"});
+              scope.userArr.unshift({id:1, name:"Core"});
             }
```
Comment (mmomen) [inline assets/js/directives/admin_top_nav_directive.js]: trailing whitespace

File: assets/js/services/auth_service.js [hunk 2]
```diff
@@ -16,11 +22,11 @@ app.factory("AuthService", ["$http", function($http) {
     .error(function() {
       return cb(null)
```
Comment (mmomen) [inline assets/js/services/auth_service.js]: missing semicolon

File: assets/js/services/bar_service.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    scope.getTeam = function(tId) {
+      $location.path("/team/"+tId);
+    };
+
+  };
+
+  return {
+    setLinks: setLinks
+  }
```
Comment (mmomen) [inline assets/js/services/bar_service.js]: missing semicolon

**Comment by mmomen** [comment]
Looks good!

File: assets/js/services/auth_service.js [hunk 4]
```diff
@@ -1,6 +1,12 @@
-app.factory("AuthService", ["$http", function($http) {
+app.factory("AuthService", ["$http", "$location", function($http, $location) {
   var user = null;
 
+  var logout = function() {
+    user = null;
+    $location.path("/login");
+    $http.get("/logout");
+  }
```
Comment (jplouie) [inline assets/js/services/auth_service.js]: ;

**Pull request merged**